src/base/EnumIterator.h

```diff
@@ -224,5 +224,20 @@ class WholeEnum : public EnumRangeT<EnumType>
     WholeEnum() : EnumRangeT<EnumType>(EnumType::enumBegin_, EnumType::enumEnd_) {}
 };
 
+/// Compile-time iteration of sequential enum values from First to Last,
+/// inclusive. The given function is called once on each iteration, with the
+/// current enum value as an argument.
+template <auto First, auto Last, class F>
+constexpr void
+ConstexprForEnum(F &&f)
+{
+    using enumType = decltype(First);
+    using underlyingType = typename std::underlying_type<enumType>::type;
+    // std::integral_constant trick makes f(First) argument a constexpr
+    f(std::integral_constant<enumType, First>()); // including when First is Last
+    if constexpr (First < Last)
+        ConstexprForEnum<enumType(static_cast<underlyingType>(First) + 1), Last>(f);
+}
```

Comment:
Nono, this is brilliant and TIL

Reply:
Please be careful with compile-time recursive. They have a tendency towards compile failure/crash on resource limited small machines/VM/containers.

Reply:
> Please be careful with compile-time recursive.

I assume this advice is not meant to trigger some specific action in this PR.

> They have a tendency towards compile failure/crash on resource limited small machines/VM/containers.

FWIW, I am aware of the risks and do not insist on this addition (because of the risks and other complications). I hope that the current use case does not run into recursion limitations due to a small number of enum values (~16), but perhaps I am too optimistic. Still, I (obviously) think it is worth trying.

Reply:
I just intended a warning to @kinkie not to get too attached to the usage.